projects/slipshow/programs/slipshow/tests/shell.nix

```diff
@@ -0,0 +1,38 @@
+{
+  sources,
+  ...
+}:
+{
+  name = "slipshow presentation";
+
+  nodes = {
+    machine =
+      { pkgs, ... }:
+      {
+        imports = [
+          sources.modules.ngipkgs
+          sources.modules.programs.slipshow
+          sources.examples.slipshow.demo-shell
+        ];
+
+        environment.etc."slipshow.md".source = pkgs.fetchurl {
+          url = "https://github.com/panglesd/slipshow/blob/v0.2.0/example/campus-du-libre/slipshow.md";
+          hash = "sha256-4+ow0GQ8RAIYDDM6rg69/X4aYs9GgqFSuEjTJmAoG8A=";
+        };
+
+      };
+  };
+
+  testScript =
+    { nodes, ... }:
+    ''
+      start_all()
+
+      # it may take around a minute to compile the file and serve it
+      machine.succeed("slipshow serve /etc/slipshow.md &")
+
+      # slipshow serves defaultly on :8080
```

Comment:
We can also add a note here that it's not possible to change the port (see https://github.com/panglesd/slipshow/issues/44).

Reply:
Oh good point.